app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/pocket/PocketStoriesComposables.kt

```diff
@@ -2,188 +2,114 @@
-            Text(
-                modifier = Modifier.padding(bottom = 2.dp),
-                text = story.publisher,
-                style = MaterialTheme.typography.caption,
-                maxLines = 1,
-                overflow = TextOverflow.Ellipsis
-            )
+    val imageUrl = story.imageUrl.replace(
+        "{wh}",
+        with(LocalDensity.current) { "${116.dp.toPx().roundToInt()}x${84.dp.toPx().roundToInt()}" }
+    )
+    ListItemTabLarge(
+        client = client,
+        imageUrl = imageUrl,
+        onClick = { onStoryClick(story) },
+        title = { TabTitle(text = story.title, maxLines = 3) },
+        subtitle = {
+            TabSubtitleWithInterdot(story.publisher, "${story.timeToRead} min")
         }
-        Text(
-            text = story.title,
-            style = MaterialTheme.typography.subtitle1,
-            maxLines = 4,
-            overflow = TextOverflow.Ellipsis
-        )
-    }
+    )
 }
 
 /**
- * Displays a list of [PocketRecommendedStory]es.
+ * Displays a list of [PocketRecommendedStory]es on 3 by 3 grid.
+ * If there aren't enough stories to fill all columns placeholders containing an external link
+ * to go to Pocket for more recommendations are added.
+ *
+ * @param stories the list of [PocketRecommendedStory]ies to be displayed. Expect a list with 8 items.
+ * @param client [Client] instance to be used for downloading the story header image.
+ * @param onExternalLinkClicked callback for when the user taps an element which contains an
+ * external link for where user can go for more recommendations.
  */
 @Composable
 fun PocketStories(
     @PreviewParameter(PocketStoryProvider::class) stories: List<PocketRecommendedStory>,
-    client: Client
+    client: Client,
+    onExternalLinkClicked: (String) -> Unit
 ) {
-    // Items will be shown on two rows. Ceil the divide result to show more items on the top row.
-    val halfStoriesIndex = (stories.size + 1) / 2
+    // Show stories in a 3 by 3 grid
+    val gridLength = 3
 
     LazyRow {
-        itemsIndexed(stories) { index, item ->
-            if (index < halfStoriesIndex) {
-                Column(
-                    Modifier.padding(end = if (index == halfStoriesIndex) 0.dp else 8.dp)
-                ) {
-                    PocketStory(item, client)
-
-                    Spacer(modifier = Modifier.height(24.dp))
+        itemsIndexed(stories.chunked(gridLength)) { rowIndex, columnItems ->
+            Column(Modifier.padding(end = if (rowIndex < gridLength - 1) 8.dp else 0.dp)) {
+                for (index in 0 until gridLength) {
+                    columnItems.getOrNull(index)?.let { story ->
+                        PocketStory(story, client) {
+                            onExternalLinkClicked(story.url)
+                        }
+                    } ?: ListItemTabLargePlaceholder(stringResource(R.string.pocket_stories_placeholder_text)) {
+                        onExternalLinkClicked("http://getpocket.com/explore")
```

Comment:
Wouldn't this cause multiple Explore cards if a column had only 1 item for example? I think you might need a `break` somewhere

Reply:
Theoretically we'd have more than enough items to display since we're also topping filtered items with ones from a `general` category.
This will soon change and the issue raised by you here would have real chances of happening so indeed, the code here needs some improvements to not always try to show a certain max number of items.